### PR TITLE
upgrade sass and grunt npm packages

### PR DESCRIPTION
**Pull request by davidsbailey**
Upgrade sass-loader and grunt-sass to eliminate the following `yarn` warning:
```
warning " > sass-loader@5.0.1" has incorrect peer dependency "webpack@^2.0.0".
```
This helps us prepare for the babel 7 upgrade.

EDIT: this also updates grunt-sass, grunt, and a few other grunt-related packages, which are a good thing to do but may not help directly with upgrading babel.

### CODE REVIEW

File: apps/Gruntfile.js [hunk 1]
```diff
@@ -303,7 +304,8 @@ describe('entry tests', () => {
       options: {
         // Compression currently occurs at the ../dashboard sprockets layer.
         outputStyle: 'nested',
-        includePaths: ['node_modules', '../shared/css/']
+        includePaths: ['node_modules', '../shared/css/'],
+        implementation: sass
```
Comment (davidsbailey) [inline apps/Gruntfile.js:308]: new required parameter in latest grunt-sass: https://github.com/sindresorhus/grunt-sass#usage

**Comment by davidsbailey** [comment]
apparently I introduced more warnings than I fixed!
```
grunt-contrib-clean@0.6.0" has incorrect peer dependency "grunt@~0.4.0".
grunt-contrib-copy@0.7.0" has incorrect peer dependency "grunt@~0.4.0".
grunt-contrib-watch@0.6.1" has incorrect peer dependency "grunt@~0.4.0".
grunt-newer@1.1.1" has incorrect peer dependency "grunt@~0.4.1".
```
I'll take a crack at upgrading these, and update the PR.

**Comment by davidsbailey** [comment]
@islemaster this is now ready to review.

**Review by islemaster — APPROVED** [review]
Pulled and ran locally.  Worked without issues!

**Pull request merged**